tendrl/gluster_integration/objects/volume/atoms/validate_shrink_volume_inputs/__init__.py

```diff
@@ -0,0 +1,221 @@
+                            for el  in brick_list:
+                                if brick_name in el:
+                                    lst = el
+                                    break
+                            else:
+                                msg = "Given brick could not be found under this volume"
+                        if brick_name not in lst:
+                            msg = "Given brick could not be found under this volume" + \
+                                  " or bricks are from different replica set"
+                            break
+                    brick_list.remove(lst)
+        # user is trying to shrink dispersed volume, check if he is providing enough bricks(k+m)
+        # for the sub-volume
+        elif disperse_count > 1:
+            sub_vol_len = disperse_count
+            for brick_set in self.parameters["Volume.bricks"]:
+                if len(brick_set) != sub_vol_len:
+                    msg = "Incorrect number of bricks provided for shrink volume" + \
+                          "Each disperse sub-vol needs %s bricks. But %s given" % (
+                              sub_vol_len, len(brick_set)
+                          )
+                    break
+                lst = []
+                for b in brick_set:
+                    brick_name = b.keys()[0] + ":" + b.values()[0].replace("/", "_")
+                    if not lst:
+                        for el  in brick_list:
+                            if brick_name in el:
+                                lst = el
+                                break
+                        else:
+                            msg = "Given brick could not be found under this volume"
+                    if brick_name not in lst:
+                        msg = "Given brick could not be found under this volume" + \
+                              " or bricks are from different sub-volumes"
+                        break
+            
+
+        if msg:
+            Event(
+                Message(
+                    priority="info",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": msg
+                    },
+                    job_id=self.parameters["job_id"],
+                    flow_id=self.parameters["flow_id"],
+                    cluster_id=NS.tendrl_context.integration_id,
+                )
+            )
+            return False
+        else:
+            Event(
+                Message(
+                    priority="info",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Inputs for shrink volume validated successfully"
+                    },
+                    job_id=self.parameters["job_id"],
+                    flow_id=self.parameters["flow_id"],
+                    cluster_id=NS.tendrl_context.integration_id,
+                )
+            )
+            return True
```

Comment:
break this entire chunk of code into functions, current code wont remain readable after a while